src/workload.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Workload manager."""
+
+from typing import Dict, Optional, Tuple
+from ops import pebble
+
+
+# A manifest is a 2-tuple of (files, layer), where "files" is a mapping from path to contents.
+# If contents is None, it means deletion.
+# TODO: contents could also be bytes
+Manifest = Tuple[Optional[Dict[str, Optional[str]]], Optional[pebble.Layer]]
```

Comment:
It's probably better for this to be a dataclass with `.files` and `.layer` -- more readable, and there might be other items in the manifest later.